### PR TITLE
Allow HTTPS connexions to sync servers

### PR DESCRIPTION
**Pull request by NDagestad**
This is a start on making the sync servers be able to use TLS.
I opened an [issue](https://github.com/KotatsuApp/kotatsu-syncserver/issues/12) about it on the sync server repo some time ago, and this is the start of what I have implemented so far.

The TLDR for the above mentioned issue, authentication is happening, and http mean it's send in the clear, wich is obviously not good.

I am not used to program in Kotlin so I wouldn't mind somebody having a look at it before I go further.

The changes are the following:
* A sync server can be specified using a full URL, including the scheme.
This means that you can specify https:// to get a secure connexion to the server. 
  * To keep some kind of backward compatibility, I chose to make http the default when nothing was specified, this way I hope things would just work for people who have already configured the sync service (but I havent tested it yet).
* The domain name validation logic got replaced with a URL validation
  * ~~it seems that the "ok" button is not disabled in the UI when the url is not valid, for some reason I have not dug into yet.~~
  * I saw that the domain validation function is also used in another part of the code, I assume that having the fallback to a domain validation will make it behave as is did before there as well but I haven't tested it, neither have I looked into why it was used somewhere else
* Omitting the scheme also does not pass the URL validation check
  * but since the check is not blocking you can still choose enter only a domain name 🙃
  * This is obviously a not a behaviour I want kept in there, maybe if the URL validation fail, the old domain validation should still be there to check if everything is good  
* A few things have been renamed to make more sense with the changes

I would also be fine with just changing the scheme to always be https, and not allow http connexions, but I though that doing such a change would be very disruptive as servers and clients would need to change together to not be to disruptive (Servers could be just put behind a revers proxy that accepts both https and http to make this more transparent for users ¯\\\_(ツ)\_/¯ )

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -375,7 +375,7 @@
 	<string name="find_similar">Find similar</string>
 	<string name="sync_settings">Synchronization settings</string>
 	<string name="server_address">Server address</string>
-	<string name="sync_host_description">You can use a self-hosted synchronization server or a default one. Don\'t change this if you\'re not sure what you\'re doing.</string>
+	<string name="sync_url_description">You can use a self-hosted synchronization server or a default one. Don\'t change this if you\'re not sure what you\'re doing.</string>
```
Comment (Koitharu) [inline app/src/main/res/values/strings.xml:378]: It`s better to not rename strings if not required

File: app/src/main/kotlin/org/koitharu/kotatsu/sync/ui/SyncHostDialogFragment.kt [hunk 2]
```diff
@@ -65,8 +65,12 @@ class SyncHostDialogFragment : AlertDialogFragment<PreferenceDialogAutocompletet
 		when (which) {
 			DialogInterface.BUTTON_POSITIVE -> {
 				val result = requireViewBinding().edit.text?.toString().orEmpty()
-				syncSettings.host = result
-				parentFragmentManager.setFragmentResult(REQUEST_KEY, bundleOf(KEY_HOST to result))
+				var scheme = ""
+				if ( ! result.startsWith("https://") && ! result.startsWith("http://")) {
```
Comment (Koitharu) [inline app/src/main/kotlin/org/koitharu/kotatsu/sync/ui/SyncHostDialogFragment.kt:69]: Please follow the project code style (remove spaces around the "`!`")

**Comment by NDagestad** [comment]
I rebased on the latest devel while fixing your comments.
Functionally speaking this should be good to go now in my opinion, unless you find something else you'd want changed 👍

File: app/src/main/res/values/strings.xml [hunk 3]
```diff
@@ -278,7 +278,7 @@
 	<string name="exclude_nsfw_from_history_summary">Manga marked as NSFW will never be added to the history and your progress will not be saved</string>
 	<string name="clear_cookies_summary">Can help in case of some issues. All authorizations will be invalidated</string>
 	<string name="show_all">Show all</string>
-	<string name="invalid_domain_message">Invalid domain</string>
+	<string name="invalid_domain_message">Invalid server address</string>
```
Comment (Koitharu) [inline app/src/main/res/values/strings.xml:281]: This string is also used for manga source domains so it would be better to create a new string resource

Comment (Koitharu) [reply app/src/main/res/values/strings.xml:281]: I`ll do it

**Pull request merged**